File: C++/ЕГЭ2021;2022/24/Решение заданий. Поляков/2715.cpp
#include <iostream>
#include <fstream>
using namespace std;

int main() {
	string s;
	ifstream f("24-j7.txt");
	f >> s;

	int max_len = 1; int len = 1;

	for (int i = 0; i < s.length(); i++) {
		int s_1 = s[i];
		int s_2 = s[i + 1];

		if ((s_1 % 2) == (s_2 % 2))
		{
			len++;
			max_len = max(len, max_len);
		}
		else {
			len = 1;
		}
	}
	cout << max_len;
	return 0;
}

File: Python/ЕГЭ2021;2022/24/Решение заданий. Поляков/2715.py
f = open('C:\\Users\\XiaoMai\\Downloads\\24-j7.txt')
s = f.readline()

max_length, length = 0, 1

for i in range(0, len(s)-1):
    if (int(s[i]) % 2 == int(s[i+1]) % 2):
        length += 1
        max_length = max(max_length, length)
    else:
        length = 1

print(max_length)
